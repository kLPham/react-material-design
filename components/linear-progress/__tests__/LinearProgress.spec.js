import React from 'react';
import { mount } from 'enzyme';
import LinearProgress from '../index';

describe('<LinearProgress />', () => {
    it('should render with progress', () => {
        const progress = 0.5;
        const linearProgress = mount(
            <LinearProgress progress={progress} />,
    );
        expect(linearProgress).toMatchSnapshot();
    });

    it('should render accent', () => {
        const accent = true;
        const progress = 0.5;
        const linearProgress = mount(
            <LinearProgress accent={accent} progress={progress} />,
    );
        expect(linearProgress).toMatchSnapshot();
    });

    it('should render indeterminate', () => {
        const indeterminate = true;
        const linearProgress = mount(
            <LinearProgress indeterminate={indeterminate} />,
    );
        expect(linearProgress).toMatchSnapshot();
    });

    it('should render indeterminate accent', () => {
        const accent = true;
        const indeterminate = true;
        const linearProgress = mount(
            <LinearProgress accent={accent} indeterminate={indeterminate} />,
    );
        expect(linearProgress).toMatchSnapshot();
    });

    it('should render indeterminate reversed', () => {
        const reversed = true;
        const indeterminate = true;
        const linearProgress = mount(
            <LinearProgress reversed={reversed} indeterminate={indeterminate} />,
        );
        expect(linearProgress).toMatchSnapshot();
    });

    it('should render with buffer', () => {
        const progress = 0.5;
        const buffer = 0.8;
        const linearProgress = mount(
            <LinearProgress progress={progress} buffer={buffer} />,
      );
        expect(linearProgress).toMatchSnapshot();
    });

    it('should render with buffer accent', () => {
        const progress = 0.5;
        const buffer = 0.8;
        const accent = true;
        const linearProgress = mount(
            <LinearProgress progress={progress} buffer={buffer} accent={accent} />,
      );
        expect(linearProgress).toMatchSnapshot();
    });

    it('should render with buffer reversed', () => {
        const progress = 0.5;
        const buffer = 0.8;
        const reversed = true;
        const linearProgress = mount(
            <LinearProgress progress={progress} reversed={reversed} buffer={buffer} />,
      );
        expect(linearProgress).toMatchSnapshot();
    });
});
