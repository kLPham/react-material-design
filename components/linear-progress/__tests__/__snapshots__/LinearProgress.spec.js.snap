// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinearProgress /> should render accent 1`] = `
<div
  className="mdc-linear-progress mdc-linear-progress--accent"
  role="progressbar"
>
  <div
    className="mdc-linear-progress__buffering-dots"
  />
  <div
    className="mdc-linear-progress__buffer"
    style={Object {}}
  />
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
    style={Object {}}
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
</div>
`;

exports[`<LinearProgress /> should render indeterminate 1`] = `
<div
  className="mdc-linear-progress mdc-linear-progress--indeterminate"
  role="progressbar"
>
  <div
    className="mdc-linear-progress__buffering-dots"
  />
  <div
    className="mdc-linear-progress__buffer"
    style={Object {}}
  />
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
    style={Object {}}
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
</div>
`;

exports[`<LinearProgress /> should render indeterminate accent 1`] = `
<div
  className="mdc-linear-progress mdc-linear-progress--accent mdc-linear-progress--indeterminate"
  role="progressbar"
>
  <div
    className="mdc-linear-progress__buffering-dots"
  />
  <div
    className="mdc-linear-progress__buffer"
    style={Object {}}
  />
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
    style={Object {}}
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
</div>
`;

exports[`<LinearProgress /> should render indeterminate reversed 1`] = `
<div
  className="mdc-linear-progress mdc-linear-progress--reversed mdc-linear-progress--indeterminate"
  role="progressbar"
>
  <div
    className="mdc-linear-progress__buffering-dots"
  />
  <div
    className="mdc-linear-progress__buffer"
    style={Object {}}
  />
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
    style={Object {}}
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
</div>
`;

exports[`<LinearProgress /> should render with buffer 1`] = `
<LinearProgress
  buffer={0.8}
  progress={0.5}
>
  <div
    className="mdc-linear-progress"
    role="progressbar"
  >
    <div
      className="mdc-linear-progress__buffering-dots"
    />
    <div
      className="mdc-linear-progress__buffer"
      style={
        Object {
          "MSTransform": "scaleX(0.8)",
          "MozTransform": "scaleX(0.8)",
          "OTransform": "scaleX(0.8)",
          "WebkitTransform": "scaleX(0.8)",
          "transform": "scaleX(0.8)",
        }
      }
    />
    <div
      className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
      style={
        Object {
          "MSTransform": "scaleX(0.5)",
          "MozTransform": "scaleX(0.5)",
          "OTransform": "scaleX(0.5)",
          "WebkitTransform": "scaleX(0.5)",
          "transform": "scaleX(0.5)",
        }
      }
    >
      <span
        className="mdc-linear-progress__bar-inner"
      />
    </div>
    <div
      className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <span
        className="mdc-linear-progress__bar-inner"
      />
    </div>
  </div>
</LinearProgress>
`;

exports[`<LinearProgress /> should render with buffer accent 1`] = `
<LinearProgress
  accent={true}
  buffer={0.8}
  progress={0.5}
>
  <div
    className="mdc-linear-progress mdc-linear-progress--accent"
    role="progressbar"
  >
    <div
      className="mdc-linear-progress__buffering-dots"
    />
    <div
      className="mdc-linear-progress__buffer"
      style={
        Object {
          "MSTransform": "scaleX(0.8)",
          "MozTransform": "scaleX(0.8)",
          "OTransform": "scaleX(0.8)",
          "WebkitTransform": "scaleX(0.8)",
          "transform": "scaleX(0.8)",
        }
      }
    />
    <div
      className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
      style={
        Object {
          "MSTransform": "scaleX(0.5)",
          "MozTransform": "scaleX(0.5)",
          "OTransform": "scaleX(0.5)",
          "WebkitTransform": "scaleX(0.5)",
          "transform": "scaleX(0.5)",
        }
      }
    >
      <span
        className="mdc-linear-progress__bar-inner"
      />
    </div>
    <div
      className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <span
        className="mdc-linear-progress__bar-inner"
      />
    </div>
  </div>
</LinearProgress>
`;

exports[`<LinearProgress /> should render with buffer reversed 1`] = `
<LinearProgress
  buffer={0.8}
  progress={0.5}
  reversed={true}
>
  <div
    className="mdc-linear-progress mdc-linear-progress--reversed"
    role="progressbar"
  >
    <div
      className="mdc-linear-progress__buffering-dots"
    />
    <div
      className="mdc-linear-progress__buffer"
      style={
        Object {
          "MSTransform": "scaleX(0.8)",
          "MozTransform": "scaleX(0.8)",
          "OTransform": "scaleX(0.8)",
          "WebkitTransform": "scaleX(0.8)",
          "transform": "scaleX(0.8)",
        }
      }
    />
    <div
      className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
      style={
        Object {
          "MSTransform": "scaleX(0.5)",
          "MozTransform": "scaleX(0.5)",
          "OTransform": "scaleX(0.5)",
          "WebkitTransform": "scaleX(0.5)",
          "transform": "scaleX(0.5)",
        }
      }
    >
      <span
        className="mdc-linear-progress__bar-inner"
      />
    </div>
    <div
      className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <span
        className="mdc-linear-progress__bar-inner"
      />
    </div>
  </div>
</LinearProgress>
`;

exports[`<LinearProgress /> should render with progress 1`] = `
<LinearProgress
  progress={0.5}
>
  <div
    className="mdc-linear-progress"
    role="progressbar"
  >
    <div
      className="mdc-linear-progress__buffering-dots"
    />
    <div
      className="mdc-linear-progress__buffer"
      style={Object {}}
    />
    <div
      className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
      style={
        Object {
          "MSTransform": "scaleX(0.5)",
          "MozTransform": "scaleX(0.5)",
          "OTransform": "scaleX(0.5)",
          "WebkitTransform": "scaleX(0.5)",
          "transform": "scaleX(0.5)",
        }
      }
    >
      <span
        className="mdc-linear-progress__bar-inner"
      />
    </div>
    <div
      className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <span
        className="mdc-linear-progress__bar-inner"
      />
    </div>
  </div>
</LinearProgress>
`;
